planex/cmd/createmockconfig.py

```diff
@@ -53,15 +53,15 @@ def load_mock_reference(fname):
     return config_opts
 
 
-def load_yum_repos(repo_names):
+def load_yum_repos(includes, excludes):
     """
     read in the yum repository configuration
     """
     yum_base = yum.YumBase()
     yum_repos = []
-    for repo_id in repo_names:
+    for repo_id in includes:
         yum_repos += yum_base.repos.findRepos(repo_id)
-    return yum_repos
+    return [rp for rp in yum_repos for ex in excludes if rp.id != ex]
```

Comment:
I'd rewrite this more declaratively
```python
return [repo for repo in yum_repos if repo.id not in excludes]
```

I would ha preferred not touch this function at all and instead modify line 145-146 below as follows:

```
enablerepo = set(args.enablerepo) - set(args.disablerepo)
yum_repos = load_yum_repos(enablerepo)
```
but this cannot work with the wildcards as used in this code.

It still feels weird to me that we need to do that, couldn't we directly pass the right list of repos in the first place instead of passing the wildcard and then a disablerepo?

Reply:
Your first rewrite suggestion is very good, thanks!
And for the question, I prefer to follow the original one which can be passed as a wildcard so that we won't have to pass a list to planex-create-mock-config, cause it may cost too much effort to enumerate the whole list items.

Reply:
So for this point, is it OK with you about such kind of configuration to parameters?
wildcard enabled for `--enablerepo`, but wildcard disabled for `--disablerepo`.

Reply:
Couldn't we do something like this?
```
set([yum_base.repos.findRepos(repo_id) for repo_id in args.enablerepo]) - set([yum_base.repos.findRepos(repo_id) for repo_id in args.disablerepo])
```

Reply:
Thanks @euanh for your comments.
And because each time calling `yum_base.repos.findRepos` it will return a list instead of item, so I made some minor changes in this code.